### DISCUSSION TITLE
UPS-RS sequence matching

### DISCUSSION THREAD

**Discussion by janossch** [dcm4che/dcm4chee-arc-light#3493] [Q&A] [answered]
Hello to everyone!

My question is related to [querying workitems](https://github.com/dcm4che/dcm4chee-arc-light/issues/2294) via UPS-RS. 

Let's imagine a workitem and let's suppose there are two items/devices in the Scheduled Station Name Code Sequence.
For sake of simplicity I made it a bit shorter:

```
{
...
 "00404025": {
            "00080100": {
                        "Value": [ "DEVICE1" ]
                    },
            "00080100": {
                        "Value": [ "DEVICE2" ]
                    },
 }
...
}
```

This query returns back the above workitem
_http://localhost:8082/dcm4chee-arc/aets/DCM4CHEE/rs/workitems?00404025.00080100=DEVICE1_

However the result of this query is **204 No Content**
_http://localhost:8082/dcm4chee-arc/aets/DCM4CHEE/rs/workitems?00404025.00080100=DEVICE2_

If I change the order of the items, then DEVICE2 value gives a match and DEVICE1 returns 204. 
It seems [sequence matching](https://dicom.nema.org/medical/dicom/current/output/html/part04.html#sect_C.2.2.2.6) is kind of not working.

Am I missing something here? Could you please help me out?
Thank you!

**Comment by vrindanayak** [comment, marked as answer]
The current code for [matching by Scheduled Station Name Code](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-query-util/src/main/java/org/dcm4chee/arc/query/util/QueryBuilder.java#L948-L949) matches only the [first item in a sequence](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-query-util/src/main/java/org/dcm4chee/arc/query/util/QueryBuilder.java#L1258-L1269) and does not fully correspond to 
> Each Item Key Attribute matching shall be performed on an Item by Item basis. 

specified in [Sequence Matching](https://dicom.nema.org/medical/dicom/current/output/html/part04.html#sect_C.2.2.2.6)

**Reply by gunterze** [reply]
Right. Will be fixed. s. #3567
